nightly-examples/examples/proxy_server.rs

```diff
@@ -84,23 +83,66 @@ async fn transfer(
     Ok(())
 }
 
+arg_enum! {
+    #[derive(PartialEq, Debug)]
+    pub enum LogFormat {
+        Plain,
+        Json,
+    }
+}
+
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     use tracing_subscriber::{EnvFilter, FmtSubscriber};
 
-    let subscriber = FmtSubscriber::builder()
-        .with_env_filter(EnvFilter::from_default_env().add_directive("proxy_server=trace".parse()?))
-        .finish();
-    tracing::subscriber::set_global_default(subscriber)?;
+    let matches = App::new("Proxy Server Exemple")
+        .version("1.0")
+        .arg(
+            Arg::with_name("log_format")
+                .possible_values(&LogFormat::variants())
+                .case_insensitive(true)
+                .long("log_format")
+                .value_name("log_format")
+                .help("Formating of the logs")
+                .required(false)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("listen_addr")
+                .long("listen_addr")
+                .help("Address to listen on")
+                .takes_value(true)
+                .required(false),
+        )
+        .arg(
+            Arg::with_name("server_addr")
+                .long("server_addr")
+                .help("Address to proxy to")
+                .takes_value(false)
+                .required(false),
+        )
+        .get_matches();
+
+    match value_t!(matches, "log_format", LogFormat).unwrap_or(LogFormat::Plain) {
```

Comment:
I would still really like it if all of this was factored out of `main` into a separate function, but it's not a blocker. It would be fine if that function called `set_global_default` so that we don't have to handle return type issues...